src/core/reducers/categories.js

```diff
@@ -96,6 +104,47 @@ export function createEmptyCategoryList(): EmptyCategoryListType {
   }, {});
 }
 
+export const getCategories = (
+  categoriesState: CategoryMapType,
+  appName: string,
+  addonType: string,
+): Array<mixed> | null => {
```

Comment:
This shouldn't be `Array<mixed>`, it should be an Array of the type you are returning.

Reply:
It should be string but when I put string in here, I am getting a flow type error 
```
Cannot return `relatedCategories` because  mixed [1] is incompatible with  string [2] in array element.
```
For removing this I used mixed.

Reply:
As mentioned above, create a new commit with all the changes I suggested (and use `string` here, like you were originally), and then I can help you debug any Flow errors you have. Flow can be quite tricky, especially when you're first getting used to it.

Reply:
Looking at this some more, this might also be the reason that you needed `mixed` in `getCategoryNames`. I think there's something wrong with this `getCategories` function.

Reply:
I think instead of using Object.values directly, I can use Object.keys as I read one of the github issue about this, also mentioned in one of the TODO [https://github.com/facebook/flow/issues/2221](https://github.com/facebook/flow/issues/2221)